### ISSUE TITLE
[gmp] Build error: building gmp:arm64-osx failed with: BUILD_FAILED

### ISSUE THREAD

**Issue by CoalBrother** [microsoft/vcpkg#33457]
### when I want to build PBC in macOS by vcpkg, something goes wrong like this

coalbrother@luhanlindeMacBook-Air vcpkg % ./vcpkg install pbc           
Computing installation plan...
pbc is only supported on '!uwp & !arm', which does not match arm64-osx. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build pbc anyway, rerun vcpkg with `--allow-unsupported`.
coalbrother@luhanlindeMacBook-Air vcpkg % ./vcpkg install pbc --allow-unsupported
Computing installation plan...
warning: pbc is only supported on '!uwp & !arm', which does not match arm64-osx. This usually means that there are known build failures, or runtime problems, when building other platforms. Proceeding anyway due to `--allow-unsupported`.
The following packages will be built and installed:
  * gmp:arm64-osx -> 6.2.1#21
    pbc:arm64-osx -> 0.5.14#8
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-osx...
Restored 0 package(s) from /Users/coalbrother/.cache/vcpkg/archives in 8.38 us. Use --debug to see more details.
Installing 1/2 gmp:arm64-osx...
Building gmp:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/coalbrother/Downloads/vcpkg/triplets/community/arm64-osx.cmake
-- Using cached gmp-6.2.1.tar.xz.
-- Cleaning sources at /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/coalbrother/Downloads/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
-- Using source at /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean
-- Getting CMake variables for arm64-osx
-- Getting CMake variables for arm64-osx-dbg
-- Getting CMake variables for arm64-osx-rel
-- Generating configure for arm64-osx
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /opt/homebrew/bin/autoreconf -vfi
    Working Directory: /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean/
    Error code: 1
    See logs for more information:
      /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/autoconf-arm64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:635 (vcpkg_execute_required_process)
  ports/gmp/portfile.cmake:85 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)

error: building gmp:arm64-osx failed with: BUILD_FAILED
Elapsed time to handle gmp:arm64-osx: 3.8 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+gmp
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?title=[gmp]+Build+error&body=Copy+issue+body+from+%2FUsers%2Fcoalbrother%2FDownloads%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

coalbrother@luhanlindeMacBook-Air vcpkg % ./vcpkg install pbc --allow-unsupported
Computing installation plan...
warning: pbc is only supported on '!uwp & !arm', which does not match arm64-osx. This usually means that there are known build failures, or runtime problems, when building other platforms. Proceeding anyway due to `--allow-unsupported`.
The following packages will be built and installed:
  * gmp:arm64-osx -> 6.2.1#21
    pbc:arm64-osx -> 0.5.14#8
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-osx...
Restored 0 package(s) from /Users/coalbrother/.cache/vcpkg/archives in 4.5 us. Use --debug to see more details.
Installing 1/2 gmp:arm64-osx...
Building gmp:arm64-osx...
warning: -- Using community triplet arm64-osx. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/coalbrother/Downloads/vcpkg/triplets/community/arm64-osx.cmake
-- Using cached gmp-6.2.1.tar.xz.
-- Cleaning sources at /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/coalbrother/Downloads/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
-- Using source at /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean
-- Getting CMake variables for arm64-osx
-- Getting CMake variables for arm64-osx-dbg
-- Getting CMake variables for arm64-osx-rel
-- Generating configure for arm64-osx
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /opt/homebrew/bin/autoreconf -vfi
    Working Directory: /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean/
    Error code: 1
    See logs for more information:
      /Users/coalbrother/Downloads/vcpkg/buildtrees/gmp/autoconf-arm64-osx-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:635 (vcpkg_execute_required_process)
  ports/gmp/portfile.cmake:85 (vcpkg_configure_make)
  scripts/ports.cmake:147 (include)

**error: building gmp:arm64-osx failed with: BUILD_FAILED
Elapsed time to handle gmp:arm64-osx: 3.8 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.

the command I tap is from https://github.com/cloudwu/pbc, in "Building pbc - Using vcpkg"

**Comment by autoantwort** [comment]
Please see here for how to report a proper issue: https://github.com/microsoft/vcpkg/issues/30604

**Comment by Cheney-W** [comment]
This issue does not reproduce on x64-osx triplet.
```
Installing 3/3 gmp:x64-osx...
Building gmp:x64-osx...
-- Downloading https://ftpmirror.gnu.org/gmp/gmp-6.2.1.tar.xz;https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz;https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz -> gmp-6.2.1.tar.xz...
-- Extracting source /Users/vcpkg/cheney/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch gmp-arm64-asm-fix-5f32dbc41afc.patch
-- Using source at /Users/vcpkg/cheney/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-osx
-- Getting CMake variables for x64-osx-dbg
-- Getting CMake variables for x64-osx-rel
-- Generating configure for x64-osx
-- Finished generating configure for x64-osx
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Installing x64-osx-dbg
-- Building x64-osx-rel
-- Installing x64-osx-rel
-- Fixing pkgconfig file: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/lib/pkgconfig/gmp.pc
-- Fixing pkgconfig file: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/lib/pkgconfig/gmpxx.pc
-- Fixing pkgconfig file: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/debug/lib/pkgconfig/gmp.pc
-- Fixing pkgconfig file: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/debug/lib/pkgconfig/gmpxx.pc
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-bases
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-fac
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-fib
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-jacobitab
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-psqr
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/manual-tools/gmp/gen-trialdivtab
-- Installing: /Users/vcpkg/cheney/vcpkg/packages/gmp_x64-osx/share/gmp/usage
-- Performing post-build validation
Elapsed time to handle gmp:x64-osx: 1.5 min
Total install time: 1.5 min
The package gmp can be imported via CMake FindPkgConfig module:

    #  gmp
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmp REQUIRED IMPORTED_TARGET gmp)
    target_link_libraries(main PkgConfig::gmp)

    # gmpxx
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmpxx REQUIRED IMPORTED_TARGET gmpxx)
    target_link_libraries(main PkgConfig::gmpxx)
```
